src/Storages/StorageS3.cpp

```diff
@@ -139,7 +139,10 @@ class StorageS3Source::DisclosedGlobIterator::Impl : WithContext
 
         request.SetBucket(globbed_uri.bucket);
         request.SetPrefix(key_prefix);
-        matcher = std::make_unique<re2::RE2>(makeRegexpPatternFromGlobs(globbed_uri.key));
+        String uri_key = globbed_uri.key;
+        if (uri_key.find(PARTITION_ID_WILDCARD) != String::npos)
+            uri_key = fmt::format("{}*", key_prefix);
```

Comment:
This doesn't seem correct.
In this implementation `some_key_{_partition_id}` is same as `some_key_*`. Why would the first be preferred over the latter?

Check `StorageURL` to see how it should be handled correctly. `{_partition_id}` should be replaced with the `partition_id` and not accept anything else.

Reply:
> This doesn't seem correct. In this implementation `some_key_{_partition_id}` is same as `some_key_*`. Why would the first be preferred over the latter?
> 
> Check `StorageURL` to see how it should be handled correctly. `{_partition_id}` should be replaced with the `partition_id` and not accept anything else.

Here, it is only used to generate the matcher of regular expression, `{_partition_id}` can be replaced by `*`, and `_file` is used to filter files.

Reply:
My point is you can already use `*`, why do we need `{_partition_id}` for the same thing?

Reply:
@antonio2368 I think `{_partition_id}` is used both read and write. when writting `{_partition_id}` will be replace to partition id, and reading `_file` will filter filename. When using `*` to write, I don't know which part of filename to replace. `*` is an unclear expression.